### DISCUSSION TITLE
Embed HLS on Wordpress Site

### DISCUSSION THREAD

**Discussion by rafastryk** [bluenviron/mediamtx#1247] [Questions]
Hi,

a newbie question: I have a small VPS running the rtsp-simple-server and a WordPress website on apache2 (mysite.com). The WP runs https using Let's Encrypt Certificate. How can I safely embed my HLS stream on this website? I tried to point the HLS stream to the same Cert as my website, but when I embed it as an iFrame my browser still says the connection is not safe, because the certificate is bound to my domain (mysite.com) not to the IP address of the server. 

`<iframe width="960" height="600" src="https://44.44.144.177:8888/stream" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>`

**Comment by aler9** [comment]
Hello, in my opinion the simplest way to solve the issue consists in using a HTTP reverse proxy to serve wordpress in a path (i.e. `/wordpress`) and rtsp-simple-server in another one (i.e. `/media-server`). You can then access any stream you want without CORS and without certificate issues.

A very popular reverse proxy with automatic Let's Encrypt support is Caddy: https://caddyserver.com/

Otherwise, if you're stuck with Apache for hosting reasons, there are reverse proxies available for Apache and even written in plain PHP, that allows to serve any website (including the one of your instance of rtsp-simple-server) through a path of your website.

[a comment by an automated account is not shown]
